core-referrals/db/migrate/20230502211749_create_core_referrers_and_referrals.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class CreateCoreReferrersAndReferrals < ActiveRecord::Migration[6.0]
+  def change
+    create_table :core_referrers, if_not_exists: true, id: :uuid do |t|
+      t.citext :code, null: false, index: { unique: true }
+      t.integer :referrals_count, default: 0
+      t.references :referrer, polymorphic: true, type: :uuid, null: false
```

Comment:
I don't love that `Core::Referrals::Referrer` would have a `referrer` association. That is confusing to me. What about `core_referral_codes` for the table?

Reply:
it would then be `core_referrer.referrer` vs `core_referral_code.code` 😐

Reply:
I also don't see it as an issue because the object is almost working as a middle man to the application specific object.